### DISCUSSION TITLE
ClientResponseError 403: with _superusers/auth-with-password

### DISCUSSION THREAD

**Discussion by SICout9010** [pocketbase/pocketbase#6191] [Q&A] [answered]
Hey everyone,
_I apologize if this can be fixed easily as I'm still beginner and this is also my first time posting a discussion._

So, I tried to authenticate as superusers on my **Production SvelteKit on Cloudflare pages + Stripe Webhook** route using 
`collection('_superusers').authWithPassword(ADMIN_SECRET_USER, ADMIN_SECRET_PASS)`
however it returns **ClientResponseError 403**, tried looking at the log and didn't found any request by _superusers not even 
`POST /api/collections/_superusers/auth-with-password` which should normally appear when doing authWithPassword()
my goal is to handle the stripe webhook to modify user's data using pocketbase after certain event type.

Here is the code(minimal handler and authenticate part) +server.ts
```JavaScript
    const adminClient = new PocketBase(PUBLIC_POCKETBASE_URL)

    export async function POST({ request }) {
      try {
          // Create pocketbase admin instance
          await adminClient.collection('_superusers').authWithPassword(ADMIN_SECRET_USER, ADMIN_SECRET_PASS).catch((err) => {
              console.log(err);
              console.log("original error: ", err.originalError);
              console.log("Failed to authenticate with Pocketbase admin client");
          });
          adminClient.autoCancellation(false);
      }
    }

```

Here is the error(on cloudflare pages) somehow this only happens on production, it works fine on localhost.
```JSON
"logs": [
    {
      "message": [
        {
          "url": "https://pburl/api/collections/_superusers/auth-with-password",
          "status": 403,
          "response": {},
          "isAbort": false,
          "originalError": {
            "url": "https://pburl/api/collections/_superusers/auth-with-password",
            "status": 403,
            "data": {}
          },
          "name": "ClientResponseError 403"
        }
      ],
      "level": "log",
      "timestamp": 1735320045522
    },
    {
      "message": [
        "original error: ",
        {
          "url": "https://pburl/api/collections/_superusers/auth-with-password",
          "status": 403,
          "data": {}
        }
      ],
      "level": "log",
      "timestamp": 1735320045522
    },
    {
      "message": [
        "Failed to authenticate with Pocketbase admin client"
      ],
      "level": "log",
      "timestamp": 1735320045522
    },
```

SvelteKit + Cloudflare pages + Stripe
OS: Windows 11
Pocketbase: v0.23.12 on Ubuntu 22.04
js-sdk: v0.24.0

_Thanks you for your help!_

**Comment by ganigeorgiev** [comment, marked as answer]
The fact that it is not reproducible locally, that the error has no standard PocketBase response and that there is no `POST ...` log in the Dashboard logs indicates that most likely the 403 response is not returned by PocketBase but by Cloudflare or any other reverse-proxy that you may be connecting through.

I'd recommend to double check your Cloudflare configuration and ensure that the everything is properly setup, e.g. [Cloudflare 4xx Client error](https://developers.cloudflare.com/support/troubleshooting/http-status-codes/4xx-client-error/#403-forbiddenrfc7231-) (_I'm not familiar with Cloudflare Pages so I'm not sure if there is anything extra that may need to be configured_).

To rule out PocketBase issue you can also try sending a direct request to your PocketBase server via curl, Postman, Insomnia, etc.:

```sh
# if PocketBase is behind Cloudflare make sure that your request is not triggering any WAF or IP rules
curl -X POST https://pburl/api/collections/_superusers/auth-with-password \
  -H 'Content-Type: application/json' \
  -d '{"identity":"test@example.com", "password":"1234567890"}'
```

**Reply by SICout9010** [reply]
YES! Thanks you so much! I never thought of Cloudflare WAF before so, I went searching for 'Stripe with Cloudflare WAF' and found this [WAF Custom Rules blocks Stripe payments](https://community.cloudflare.com/t/waf-custom-rules-blocks-stripe-payments/536520) 

eventually I also found out that I have **Bot Fight Mode On** in Security > Bots, what I did is that I **turn it off** and resend the request from Stripe again, then it work. 🎉
